File: old/problem0686/log_try.cc
#include <iostream>
#include <cmath>
#include <climits>

using namespace std;

int main(int arc, char **argv)
{
  double log_2 = log10(2.);

  int n_max = atol(argv[1]);

  int p = 1;
  int n = 0;
  double log_lo = log10(1.23);
  double log_hi = log10(1.24);
    
  double log_n,log_m,intp;
    
  while (n < n_max) {
    double log_n = p * log_2;
    double log_m = modf(log_n,&intp);
    if ((log_m >= log_lo) && (log_m < log_hi)){
      n++;
      if (n%10000 == 0) {
	cout << "." << flush;
      }
    }
    p++;
  }

  cout << endl << n << " " << p - 1 << endl;
  
}

File: old/problem0686/log_try.py
#!/usr/bin/env python

from math import log10
import sys

log_2 = log10(2)

log_lo = log10(1.23)
log_hi = log10(1.24)


for max_n in [int(s) for s in sys.argv[1:]]:

    p = 1
    n = 0
    while n < max_n: 
        log_n = p*log_2
        log_m = log_n - int(log_n)
        if (log_m >= log_lo) and (log_m < log_hi):
            n += 1
            if n%10000 == 0:
                sys.stdout.write('.')
                sys.stdout.flush()

        p += 1
            

    print("\n",n,p-1)

